cmd/appgw-ingress/main.go

```diff
@@ -161,14 +162,20 @@ func main() {
 
 	if err = azure.WaitForAzureAuth(azClient, maxAuthRetryCount, retryPause); err != nil {
 		if err == azure.ErrAppGatewayNotFound && env.EnableDeployAppGateway {
-			if env.AppGwSubnetID != "" {
-				err = azClient.DeployGatewayWithSubnet(env.AppGwSubnetID)
-			} else if azContext != nil {
-				err = azClient.DeployGatewayWithVnet(azure.ResourceGroup(azContext.VNetResourceGroup), azure.ResourceName(azContext.VNetName), azure.ResourceName(env.AppGwSubnetName), env.AppGwSubnetPrefix)
-			}
+			// create application gateway with retries
+			err = utils.Retry(maxAuthRetryCount, retryPause,
+				func() (utils.Retriable, error) {
+					if env.AppGwSubnetID != "" {
+						err = azClient.DeployGatewayWithSubnet(env.AppGwSubnetID)
+					} else if azContext != nil {
+						err = azClient.DeployGatewayWithVnet(azure.ResourceGroup(azContext.VNetResourceGroup), azure.ResourceName(azContext.VNetName), azure.ResourceName(env.AppGwSubnetName), env.AppGwSubnetPrefix)
+					}
+					glog.Error("Failed in deploying Application Gateway and dependencies: ", err)
```

Comment:
I think we should separate the deployment of VNet and AppGw. Retrying on failed appgw deployment doesnot make sense to me . 

We should separate it out so that deployment of appgw failing doesnot intervene with aks cluster coming up ? What do you think